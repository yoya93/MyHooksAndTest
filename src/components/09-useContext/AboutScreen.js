import React from "react";

export const AboutScreen = () => {
  return (
    <div>
      <h1>AbautScreen</h1>
      <hr />
    </div>
  );
};
