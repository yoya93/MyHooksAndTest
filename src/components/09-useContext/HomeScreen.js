import React from "react";
import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  const { name } = user;

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
      <h3>{name}</h3>
    </div>
  );
};
