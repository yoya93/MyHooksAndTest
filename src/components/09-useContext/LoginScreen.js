import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        onClick={() => {
          setUser({
            name: "jorge",
          });
        }}
        className="btn btn-primary"
      >
        Login
      </button>
    </div>
  );
};
