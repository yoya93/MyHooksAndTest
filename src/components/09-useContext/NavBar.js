import React from "react";
import { Link, NavLink } from "react-router-dom";

//NavLink establece  una clase si el url coincide con el path

export const NavBar = () => {
  return (
    <div>
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <Link className="navbar-brand" to="/">
          UserContext
        </Link>

        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link "
              to="/"
            >
              Home <span className="sr-only">(current)</span>
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="/login"
            >
              Login
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="/about"
            >
              About
            </NavLink>
          </div>
        </div>
      </nav>
    </div>
  );
};
