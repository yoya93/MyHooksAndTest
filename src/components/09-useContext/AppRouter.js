import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { AboutScreen } from "./AboutScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />

        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Route exact path="/about" component={AboutScreen} />
          </Switch>
        </div>
      </div>
    </Router>
  );
};
