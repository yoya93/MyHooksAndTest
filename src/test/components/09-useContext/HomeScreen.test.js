import React from "react";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { act } from "@testing-library/react";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en <HomeScreen /> ", () => {
  const user = {
    name: "jorge",
    email: "jorhe@gmail.com",
  };
  //   const wrapper = shallow(//con este solamente se muestra el hoc po lo tanto no me sirve ya que necesito ver como se renderizo el mount
  //     <UserContext.Provider>
  //       <HomeScreen user={user} />
  //     </UserContext.Provider>
  //   );

  const wrapper = mount(
    //con este solamente se muestra el hoc po lo tanto no me sirve ya que necesito ver como se renderizo el mount
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test("debe de mostrarse correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
