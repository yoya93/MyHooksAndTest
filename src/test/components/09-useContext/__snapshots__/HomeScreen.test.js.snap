// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen />  debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "jorge",
   "email": "jorhe@gmail.com"
}
  </pre>
  <h3>
    jorge
  </h3>
</div>
`;
